frontend/src/scratch/Scratch.tsx

```diff
@@ -298,7 +310,25 @@ export default function Scratch() {
                                 <Skeleton height={20} count={20} />
                             </> : <>
                                 {(showWarnings || !diff) && <code class={styles.log}>{log}</code>}
-                                <code class={styles.diff} dangerouslySetInnerHTML={{ __html: diff }} />
+                                {(diff && diff.error) && <code class={styles.log}>{diff.error}</code>}
+                                {(diff && !diff.error) &&
+                                    <>
+                                        <table class={styles.diff}>
+                                            <tr>
+                                                <th>{formatDiffText(diff.header.base)}</th>
+                                                <th>{/* Line */}</th>
```

Comment:
Why not show the line no here?

Reply:
This is for the header row (`TARGET`, `CURRENT (100)`) -- since the line column is narrow, I didn't want to add a label at the top? (Unless you think I should? None of the other "sub columns" are labelled, like the offset)

Reply:
Ah, I didn't realise this was the header row!

Although, thinking about it, it might be nice to explain what each "sub column" is, if there's space.